[rpc] fix RRef alias annotation

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#39933 [rpc] fix RRef alias annotation**
* #39932 [rpc] use annotation_str for RRef type serialization

Fix rref related alias annotation to ensure it's not getting ebased by
the jit dce.

Differential Revision: [D22015426](https://our.internmc.facebook.com/intern/diff/D22015426)

wanchaol committed:
[rpc] fix RRef alias annotation

Fix rref related alias annotation to ensure it's not getting ebased by
the jit dce.

[ghstack-poisoned]

mrshenli commented on torch/csrc/jit/runtime/register_distributed_ops.cpp:
```diff
@@ -26,7 +26,7 @@ static auto workerInfo =
 RegisterOperators reg_rpc_ops(
     {Operator(
-         "aten::to_here(RRef(t) self) -> t",
+         "aten::to_here(RRef(t) self) -> t(*)",
```
Curious, do we also change `t` to `t(*)` in `RRef(t)`?

wanchaol replied:
not needed, the change to the return type is to ensure the return value is aliased to any other values so that the dce will not remove this op, the `RRef(t)` does not need to alias to any other value like the return value

mrshenli reviewed (approved):
Thanks for fixing promptly!

(Not suggesting changes in this PR) Do we also need to fix `Future(t)` here? https://github.com/pytorch/pytorch/blob/193bf6ed5970efd0c981783579141e26ba3fc79b/torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp#L675

wanchaol committed:
Update on "[rpc] fix RRef alias annotation"


Fix rref related alias annotation to ensure it's not getting ebased by
the jit dce.

Differential Revision: [D22015426](https://our.internmc.facebook.com/intern/diff/D22015426)

[ghstack-poisoned]

wanchaol commented:
> Thanks for fixing promptly!
> 
> (Not suggesting changes in this PR) Do we also need to fix `Future(t)` here?
> 

good point, `wait` is a bit complicated with special casing and side effects, I will try to make it consistent with a follow up PR.

wanchaol committed:
Update on "[rpc] fix RRef alias annotation"


Fix rref related alias annotation to ensure it's not getting ebased by
the jit dce.

Differential Revision: [D22015426](https://our.internmc.facebook.com/intern/diff/D22015426)

[ghstack-poisoned]
